src/Compilers/CSharp/Portable/Lowering/Instrumentation/DebugInfoInjector.cs

```diff
@@ -140,39 +142,41 @@ public override BoundStatement InstrumentYieldReturnStatement(BoundYieldReturnSt
             return AddSequencePoint(base.InstrumentYieldReturnStatement(original, rewritten));
         }
 
-        public override BoundStatement? CreateBlockPrologue(BoundBlock original, out Symbols.LocalSymbol? synthesizedLocal)
+        public override void InstrumentBlock(BoundBlock original, LocalRewriter rewriter, ref TemporaryArray<LocalSymbol> additionalLocals, out BoundStatement? prologue, out BoundStatement? epilogue)
         {
-            var previous = base.CreateBlockPrologue(original, out synthesizedLocal);
-            if (original.Syntax.Kind() == SyntaxKind.Block && !original.WasCompilerGenerated)
-            {
-                var oBspan = ((BlockSyntax)original.Syntax).OpenBraceToken.Span;
-                return new BoundSequencePointWithSpan(original.Syntax, previous, oBspan);
-            }
-            else if (previous != null)
-            {
-                return new BoundSequencePoint(original.Syntax, previous);
-            }
+            base.InstrumentBlock(original, rewriter, ref additionalLocals, out var previousPrologue, out var previousEpilogue);
 
-            return null;
-        }
-
-        public override BoundStatement? CreateBlockEpilogue(BoundBlock original)
-        {
-            var previous = base.CreateBlockEpilogue(original);
+            prologue = previousPrologue;
+            epilogue = previousEpilogue;
 
-            if (original.Syntax.Kind() == SyntaxKind.Block && !original.WasCompilerGenerated)
+            if (original.Syntax is BlockSyntax blockSyntax && !original.WasCompilerGenerated)
             {
+                prologue = new BoundSequencePointWithSpan(original.Syntax, previousPrologue, blockSyntax.OpenBraceToken.Span);
+
                 // no need to mark "}" on the outermost block
                 // as it cannot leave it normally. The block will have "return" at the end.
                 SyntaxNode? parent = original.Syntax.Parent;
                 if (parent == null || !(parent.IsAnonymousFunction() || parent is BaseMethodDeclarationSyntax))
                 {
-                    var cBspan = ((BlockSyntax)original.Syntax).CloseBraceToken.Span;
-                    return new BoundSequencePointWithSpan(original.Syntax, previous, cBspan);
+                    epilogue = new BoundSequencePointWithSpan(original.Syntax, previousEpilogue, blockSyntax.CloseBraceToken.Span);
                 }
             }
+            else if (original == rewriter.CurrentMethodBody)
+            {
+                if (previousPrologue != null)
+                {
+                    prologue = BoundSequencePoint.CreateHidden(previousPrologue);
+                }
+                else if (rewriter.Factory.TopLevelMethod is SynthesizedSimpleProgramEntryPointSymbol)
+                {
+                    prologue = BoundSequencePoint.CreateHidden();
+                }
```

Comment:
What is the impact of this addition? Does this fix a bug?

Reply:
Is this just a move of code from `LocalRewriter.VisitBlock()`?

Reply:
Yes, just move.